pkg/timer_event/cleanup_users_marked_for_deletion.go

```diff
@@ -0,0 +1,96 @@
+package timer_event
+
+import (
+	"context"
+
+	"github.com/coneno/logger"
+	"github.com/influenzanet/go-utils/pkg/api_types"
+	"github.com/influenzanet/go-utils/pkg/constants"
+	loggingAPI "github.com/influenzanet/logging-service/pkg/api"
+	messageAPI "github.com/influenzanet/messaging-service/pkg/api/messaging_service"
+	"github.com/influenzanet/user-management-service/pkg/utils"
+)
+
+// CleanupUsersMarkedForDeletion handles the deletion of accounts that did not react to reminder mail
+func (s *UserManagementTimerService) CleanupUsersMarkedForDeletion() {
+	logger.Debug.Println("Starting clean up job for users marked for deletion:")
+	instances, err := s.globalDBService.GetAllInstances()
+	if err != nil {
+		logger.Error.Printf("unexpected error: %s", err.Error())
+	}
+	for _, instance := range instances {
+		users, err := s.userDBService.FindUsersMarkedForDeletion(instance.InstanceID)
+		count := 0
+
+		if err != nil {
+			logger.Error.Printf("unexpected error: %s", err.Error())
+			continue
+		}
+		for _, u := range users {
+			// ---> Trigger message sending
+			_, err = s.clients.MessagingService.SendInstantEmail(context.TODO(), &messageAPI.SendEmailReq{
+				InstanceId:        instance.InstanceID,
+				To:                []string{u.Account.AccountID},
+				MessageType:       constants.EMAIL_TYPE_ACCOUNT_DELETED_AFTER_INACTIVITY,
+				PreferredLanguage: u.Account.PreferredLanguage,
+				UseLowPrio:        true,
+			})
+			if err != nil {
+				logger.Error.Printf("DeleteAccount: %s", err.Error())
+			}
+			err := s.globalDBService.DeleteAllTempTokenForUser(instance.InstanceID, u.ID.Hex(), "")
+			if err != nil {
+				logger.Error.Printf("error, when trying to remove temp-tokens: %s", err.Error())
+				continue
+			}
+			_, err = s.userDBService.DeleteRenewTokensForUser(instance.InstanceID, u.ID.Hex())
+			if err != nil {
+				logger.Error.Printf("error, when trying to remove renew tokens: %s", err.Error())
+				continue
+			}
+			err = s.userDBService.DeleteUser(instance.InstanceID, u.ID.Hex())
+			if err != nil {
+				logger.Error.Printf("error, when trying to delete user: %s", err.Error())
+				continue
+			}
+
+			//notify study service
+			mainProfileID, otherProfileIDs := utils.GetMainAndOtherProfiles(u)
```

Comment:
Do you know if study service returns an error if participant does not exists or similar cases? I was thinking, that it might be better, to not delete user yet, if study service cannot be notified. In case, there is a temporary network issue, or similar, we would loose the info. It might be better, to delete the user only if study service method could be run for all of the profile. If any failed, keep user, and try to delete again. (We might even need to check what the error message is, to accept some error, but not other errors)
With the same motivation, it would probably be better in this case, to notify user about deletion only if the deletion really happened and not before.

Reply:
I switched process around and it works. User is now deleted only if study service notification was successful. Please check.